Core/Core/CourseSync/Common/Models/Entities/CourseSyncEntry+Array.swift

```diff
@@ -0,0 +1,71 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2023-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Foundation
+
+public extension Array where Element == CourseSyncEntry {
+    var totalSelectedSize: Int {
+        reduce(0) { partialResult, entry in
+            partialResult + entry.totalSelectedSize
+        }
+    }
+
+    var totalDownloadedSize: Int {
+        reduce(0) { partialResult, entry in
+            partialResult + entry.totalDownloadedSize
+        }
+    }
+
+    var progress: Float {
+        Float(totalDownloadedSize) / Float(totalSelectedSize)
```

Comment:
Could we have a zero check here and at the other instances of divisions just to be safe?

Reply:
Fixed.